Engine Nightly Checks fixes

hubertp opened the pull request:
### Pull Request Description

An attempt to fix remaining failures in our extra tests running nightly.
Partial revert of #11805 was needed as some of our tests use the dev version and broke in the process.
Regular CI builds didn't complain, apparently, since the problematic test used the "skip-on-failure" exception but locally it was problematic.

### Important Notes

This is an attempt to make [Engine Nightly Checks](https://github.com/enso-org/enso/actions/workflows/engine-checks-nightly.yml) pass.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] Unit tests have been written where possible.

hubertp committed:
Partial fixes to nightly tests

An attempt to fix remaining failures in our extra tests running nightly.

hubertp committed:
Stop skipping on failure

Will probably start failing on CI, as it should to begin with.

hubertp commented:
So, remaining issues in `LibrariesTest` are due to https://github.com/enso-org/enso/pull/11805, which seems to discard `0.0.0-dev` editions. Those are needed in our tests.

hubertp committed:
Partial revert of #11805

PR #11805 removed support for storing `0.0.0-dev` versions. Sadly it
went unnoticed that it broke our test-suite. Bringing it back,
optionally, so that we can keep using it in our tests.

hubertp commented:
Now
```
sbt:enso> language-server/testOnly org.enso.languageserver.websocket.json.LibrariesTest
``` 
will pass.

hubertp committed:
nit

hubertp committed:
Stabilize Table_Visualization tests

If data quality metrics are still pending for distinct count, it would
return `Nothing`. This in turn meant that our tests in `Table_Spec`
could be flaky as `is_using_multi_filter` would return `False` in that
case. This was more visible in our native image builds as things were
faster.
Not sure why the behaviour was changed in #13583.

hubertp commented:
The last commit in https://github.com/enso-org/enso/pull/13781/commits/90438a3f216c1321097e3aa05dc5afc9ee948d58 stabilizes the results. Not sure why @jdunkerley  changed it in #13583 so please take a look.

No more
```diff
    - [FAILED] should visualize database columns [405ms]
        Reason: '{"header":["Average C"],"value_type":[{"constructor":"Float","display_text":"Float (64 bits)"}],"all_rows_count":2,"has_index_col":true,"get_child_node_action":"at","use_bottom_status_bar":false,"enable_create_node":false,"data_quality_metrics":[],"type":"EnsoTableOrColumn","child_label":"row","is_using_server_sort_and_filter":false,"requires_number_format":[false],"table_version_hash":null,"is_using_multi_filter":[false],"data":[[4.0]]}' did not equal '{"header":["Average C"],"value_type":[{"constructor":"Float","display_text":"Float (64 bits)"}],"all_rows_count":2,"has_index_col":true,"get_child_node_action":"at","use_bottom_status_bar":false,"enable_create_node":false,"data_quality_metrics":[],"type":"EnsoTableOrColumn","child_label":"row","is_using_server_sort_and_filter":false,"requires_number_format":[false],"table_version_hash":null,"is_using_multi_filter":[true],"data":[[4.0]]}' (at src/Table_Spec.enso:97:13-37).
```
which I even got locally ocassionally.

JaroslavTulach commented on lib/scala/pkg/src/test/scala/org/enso/pkg/ConfigSpec.scala:
```diff
@@ -103,8 +104,8 @@ class ConfigSpec
 
-      val ser = parsed.toYaml
-      ser shouldEqual "name: fooBar\nnamespace: local\n"
+      val ser = parsed.toYaml(keepDevVersions = true)
```
Shouldn't this test be duplicated? Once more with `false`?

hubertp replied:
Good point, yes.

JaroslavTulach commented on lib/scala/library-manager/src/test/scala/org/enso/librarymanager/test/published/repository/DummyRepository.scala:
```diff
@@ -97,7 +97,7 @@ abstract class DummyRepository(toolsRootDirectory: Path) {
       version   = lib.version.toString()
     )
-    pkg.save()
+    pkg.save(true)
```
What was the failure when `0.0.0-dev` wasn't stored into the `package.yaml`?

hubertp replied:
No editions found. Feel free to run `LibrariesTest` locally as everyone seemed to have ignored it.

JaroslavTulach replied:
I see.
```
sbt language-server/testOnly *LibrariesTest
```
fails on my computer. I saw the failure before, but never on the CI for my PRs, so I assumed it is some build mismatch problem.

JaroslavTulach commented on lib/scala/pkg/src/test/scala/org/enso/pkg/ConfigSpec.scala:
```diff
@@ -36,7 +36,8 @@ class ConfigSpec
         services             = List(),
         jvm                  = None
       )
-      val deserialized = Config.fromYaml(config.toYaml).get
```
Were these tests not executed in the CI? I'm still trying to understand what #11805 could break and why that's bad? E.g. why not fix the test?

JaroslavTulach replied:
I saw the `LibrariesTest` failure before locally, but never on the CI for my PRs, so I assumed it is some build mismatch problem.

- why doesn't it fail the CI for my PRs?
- is `language-server/test` executed?
- if so, does it succeeds because CI for the PRs sets version to 2025.x.y?
- can't we also set a version to 2025.x.y for the nightly checks then?

JaroslavTulach commented on engine/language-server/src/test/scala/org/enso/languageserver/websocket/json/BaseServerTest.scala:
```diff
@@ -448,8 +448,9 @@ abstract class BaseServerTest
-        edition = customEdition
+        name            = "TestProject",
+        edition         = customEdition,
+        keepDevVersions = true
```
Is this the test that was broken by #11805?

JaroslavTulach replied:
The failing test is `LibrariesTest` from `language-server` project.

JaroslavTulach reviewed (commented):
- I consider #11805 behavior correct
- why don't we fix the tests instead of duplicating the code...
- ... and then testing something that isn't usable in production?

hubertp commented:
@JaroslavTulach To answer your suggestion on "why don't we fix the tests instead of duplicating the code...":
This is not as easy it sounds, I'm afraid. The libraries infrastructure uses the current build version to push editions. So... either way some changes would have to be hacked so that we don't do that in _non-test_ code. Out of the two options I had I chose the one which I felt was the least intrusive.
After all `0.0.0-dev` is a version that we support in regular dev workflow and it is here to stay (that's why I had objections about #11805 in the first place but I'm not going insist on reverting it completely).

hubertp committed:
Fix module path for editions

If the edition was missing users would get messages along the line of
```
ava.lang.IllegalAccessError: class org.enso.editions.updater.EditionUpdater (in module org.enso.editions.updater) cannot access class org.enso.downloader.http.URIBuilder (in module org.enso.downloader) because module org.enso.editions.updater does not read module org.enso.downloader
	at org.enso.editions.updater/org.enso.editions.updater.EditionUpdater.$anonfun$fetchManifests$2(EditionUpdater.scala:100)
```
which is bad.

hubertp committed:
Add missing test

hubertp committed:
Remove spurious asserts

`assertSameElements` that follows already checks the size of the
responses **and** provides a more useful error - the messages that have
been received. Without the latter, there is little one can do to debug
the issue on CI.

hubertp commented:
I'm hoping that https://github.com/enso-org/enso/pull/13781/commits/c5109b0d08615ded682183fe11afb5480d05b09c will help with debugging spurious errors like https://github.com/enso-org/enso/actions/runs/17074739725/job/48472472179?pr=13781#step:14:5694

<blockquote><img src="https://opengraph.githubassets.com/2a66e1cd157c9e9dcbe49200ab5cf74a04be39788b751af693190ca90dddff20/enso-org/enso" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/enso-org/enso">Engine Nightly Checks fixes · enso-org/enso@e5b01cf</a></strong></div><div>Enso Analytics is a self-service data prep and analysis platform designed for data teams. - Engine Nightly Checks fixes · enso-org/enso@e5b01cf</div></blockquote>

hubertp commented:
Merging as MacOS problems appear unrelated and we need that vis fix for flaky tests.
